File: spring_java_orm_sample/src/main/java/com/example/demo/api/response/PurchaseInvoiceResponse.java
package com.example.demo.api.response;

import com.example.demo.data.domain.PurchaseInvoiceType;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.Value;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

@Value
public class PurchaseInvoiceResponse {
    @Schema(description = "仕入伝票ID", type = "integer", format = "int64")
    @NotNull
    Long id;

    @Schema(description = "仕入伝票種別")
    @NotNull
    PurchaseInvoiceType purchaseInvoiceType;

    @Schema(description = "返品元仕入伝票ID", type = "integer", format = "int64")
    Long returnPurchaseInvoiceId;

    @Schema(description = "仕入伝票日付", type = "string", format = "date")
    @NotNull
    LocalDate purchaseInvoiceDate;

    @Schema(description = "仕入先ID", type = "integer", format = "int64")
    @NotNull
    Long supplierId;

    @Schema(description = "入庫店舗ID", type = "integer", format = "int64")
    @NotNull
    Long receivingStoreId;

    @Schema(description = "仕入伝票金額", type = "integer", format = "int64")
    @NotNull
    Long purchaseInvoiceAmount;

    @Schema(description = "更新日時", type = "string", format = "date-time")
    @NotNull
    LocalDateTime updateAt;

    @Schema(description = "バージョン", type = "integer", format = "int64")
    @NotNull
    Long version;

    @Schema(description = "仕入伝票明細")
    @NotEmpty
    List<PurchaseInvoiceDetailResponse> detail;
}

File: codex_poting/kotlin/src/main/kotlin/com/example/demo/api/response/PurchaseInvoiceResponse.kt
package com.example.demo.api.response

import com.example.demo.data.domain.PurchaseInvoiceType
import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotEmpty
import jakarta.validation.constraints.NotNull
import java.time.LocalDate
import java.time.LocalDateTime

class PurchaseInvoiceResponse {
    @field:Schema(description = "仕入伝票ID", type = "integer", format = "int64")
    @field:NotNull
    var id: Long? = null

    @field:Schema(description = "仕入伝票種別")
    @field:NotNull
    var purchaseInvoiceType: PurchaseInvoiceType? = null

    @field:Schema(description = "返品元仕入伝票ID", type = "integer", format = "int64")
    var returnPurchaseInvoiceId: Long? = null

    @field:Schema(description = "仕入伝票日付", type = "string", format = "date")
    @field:NotNull
    var purchaseInvoiceDate: LocalDate? = null

    @field:Schema(description = "仕入先ID", type = "integer", format = "int64")
    @field:NotNull
    var supplierId: Long? = null

    @field:Schema(description = "入庫店舗ID", type = "integer", format = "int64")
    @field:NotNull
    var receivingStoreId: Long? = null

    @field:Schema(description = "仕入伝票金額", type = "integer", format = "int64")
    @field:NotNull
    var purchaseInvoiceAmount: Long? = null

    @field:Schema(description = "更新日時", type = "string", format = "date-time")
    @field:NotNull
    var updateAt: LocalDateTime? = null

    @field:Schema(description = "バージョン", type = "integer", format = "int64")
    @field:NotNull
    var version: Long? = null

    @field:Schema(description = "仕入伝票明細")
    @field:NotEmpty
    var detail: List<PurchaseInvoiceDetailResponse>? = null
}
